### DISCUSSION TITLE
Current Status #10March

### DISCUSSION THREAD

**Discussion by smbanaru** [banarutz/BNRNet#10] [General]
All the metrics are calculated on the validation set which contains 1k images with the same distribution as the training set.
Currently, I trained 3 models:
- VGG16 (image size 178 x 218) 
- ResNet50 (image size 192 x 192)
- Resnet50 using dropout and batch norm regularization (image size 192 x 192)

All the models are selected with the lowest validation MAE score which is also the loss function that I used. The arhitectures are Encoder-Decoder because both of them (VGG and ResNet) are downsampling the images in their forward pass so I used transposed convolutions to upsample them.

## VGG16 (epoch 50)

**MSE**: 1249.703
**MAE**: 24.503
**SNR**: 12.179
**PSNR**: 18.088
**SSIM**: 0.859 (closer to 1 is better)
**LPIPS**: 0.126 (closer to 0 is better)

## ResNet50 (epoch 21)
**MSE**: 1270.019
**MAE**: 24.839
**SNR**: 12.073
**PSNR**: 17.952
**SSIM**: 0.85
**LPIPS**: 0.127

Training tensorboard:

![signal-2023-03-09-161755_002](https://user-images.githubusercontent.com/122101840/224356334-2884ff3c-e840-4f3a-8fa9-c5760e0ed5ac.png)

## ResNet50 V2
- to be updated but the training tensorboard not looking fine

![signal-2023-03-09-161730_002](https://user-images.githubusercontent.com/122101840/224356437-7b78a354-c46d-46e1-ab6a-510f6a74852b.png)

## Visual comparasion
-------------------------------------

### From my POV ResNet images looks better.

![000001](https://user-images.githubusercontent.com/122101840/224358307-786f001c-17cc-4380-99e5-14665ba6c2f4.png)

![000002](https://user-images.githubusercontent.com/122101840/224358318-b772ae5e-c2c3-427b-9ecf-eaf76f5983d6.png)

![000003](https://user-images.githubusercontent.com/122101840/224358336-5e821c46-6cc2-498e-b632-0dcfb6d18f7c.png)

![000004](https://user-images.githubusercontent.com/122101840/224358396-0fb71147-7262-4f9f-9e5e-9bfe87b0161b.png)

![000005](https://user-images.githubusercontent.com/122101840/224358426-18783303-b0a1-4754-a7db-cf261273b8ea.png)

![000006](https://user-images.githubusercontent.com/122101840/224358450-ee430de1-7db5-4321-bfc0-38e1d3848fdf.png)

![000007](https://user-images.githubusercontent.com/122101840/224358480-b4197abd-6d10-4948-bdcc-74c943f698f3.png)
![000008](https://user-images.githubusercontent.com/122101840/224358489-3ba36e40-fea3-45b1-b1ae-1d60b8443377.png)
![000009](https://user-images.githubusercontent.com/122101840/224358494-9f10d439-6bbc-4189-9cfb-d11d9a18bc66.png)
![000010](https://user-images.githubusercontent.com/122101840/224358496-de07ccc4-be59-44bf-b32b-002455668199.png)
![000011](https://user-images.githubusercontent.com/122101840/224358498-b53ce432-9a28-4089-80a6-4f2ce0a99b1c.png)
![000012](https://user-images.githubusercontent.com/122101840/224358499-9d3675c5-4fa2-48b2-9197-f5947978f1a9.png)
![000013](https://user-images.githubusercontent.com/122101840/224358504-8d2c0c36-709f-4438-a036-0c7f9b314dfc.png)
![000014](https://user-images.githubusercontent.com/122101840/224358506-889a3f5f-623c-49ff-9c34-c2b4922ad0dc.png)
![000015](https://user-images.githubusercontent.com/122101840/224358508-cd3401f1-6786-49ad-93a7-69fcc6d0a564.png)
![000016](https://user-images.githubusercontent.com/122101840/224358510-ed24c7fb-2122-4dbd-a70c-7b47fb6925de.png)
![000017](https://user-images.githubusercontent.com/122101840/224358512-22b77fd3-75a3-4eed-bbed-a10e87250728.png)
![000018](https://user-images.githubusercontent.com/122101840/224358516-6d3640c9-b00e-4d93-acaa-84da905907c5.png)

I will compare make the inference using SOTA models in the near future but I was struggling for 2hrs to make it work today with no success:))

Currently is training a ResNet50 with MSE as a loss function.

## TODO

- I want to try @eduardatmadenn idea to use perceptual loss as a part of the whole loss.

- Also, I have to present the kNN method and the matlab method too.

**Comment by smbanaru** [comment]
# SOTA comparasion - [Disentangled Colorization](https://github.com/ToTheBeginning/GCP-Colorization)

I used the beta model and the results has very high saturation and it seems that is not suitable for portraits. For example:
![000029](https://user-images.githubusercontent.com/122101840/225578175-8370ea52-3587-4857-b81b-88211a064bee.png)

With the other model, c0.2 model I had the following results:

**MSE**: 625.55
**MAE**: 14.38
**SNR**: 16.764
**PSNR**: 22.644
**SSIM**: 0.916
**LPIPS**: 0.117

## Examples:
------------------------
![000001](https://user-images.githubusercontent.com/122101840/225581884-8ffb4360-3abe-4aa4-87cc-4c274d72d89f.png)
![000002](https://user-images.githubusercontent.com/122101840/225581898-11bb40aa-2af0-4253-a187-aab66123f0b9.png)
![000003](https://user-images.githubusercontent.com/122101840/225581902-05d984d5-569d-4707-a9f5-3f4ce02671fe.png)
![000004](https://user-images.githubusercontent.com/122101840/225581907-17ccf0b3-2a30-42b4-b9f8-9b06d914e3df.png)
![000005](https://user-images.githubusercontent.com/122101840/225581911-67899d88-87e5-4ffb-890c-18d231c7c93a.png)
![000006](https://user-images.githubusercontent.com/122101840/225581916-6e22dbbf-79d7-4927-b620-1b977704dc84.png)
![000007](https://user-images.githubusercontent.com/122101840/225581924-ae237a8f-be8f-4732-b5c1-cf50eed796a9.png)
![000008](https://user-images.githubusercontent.com/122101840/225581928-9121d47a-db8b-4f8d-96e8-2236035b6cf3.png)
![000009](https://user-images.githubusercontent.com/122101840/225581935-829e5339-516a-4bcc-bd69-5ed6ba49a73c.png)
![000010](https://user-images.githubusercontent.com/122101840/225581946-132f2c38-e97a-4037-991b-6bc373b9da98.png)
![000011](https://user-images.githubusercontent.com/122101840/225581953-3d1f473e-bdbb-4f52-b5e3-51bd543758f8.png)
![000012](https://user-images.githubusercontent.com/122101840/225581959-426bbdb9-0772-40aa-8e0d-dbdf51f6d7cd.png)
![000013](https://user-images.githubusercontent.com/122101840/225581963-7c411bd6-ba8f-429e-a061-99be25bfaddd.png)
![000014](https://user-images.githubusercontent.com/122101840/225581965-5e991112-fa23-4482-9947-3764e4677fa3.png)
![000015](https://user-images.githubusercontent.com/122101840/225581969-4f91988b-18f4-4a9c-9c38-2049da696774.png)
![000016](https://user-images.githubusercontent.com/122101840/225581974-adfaa7a1-94c0-4bdc-a7dd-08dfff826a4f.png)
![000017](https://user-images.githubusercontent.com/122101840/225581979-3e421de6-2f59-4687-82d5-098ed7305c70.png)
![000018](https://user-images.githubusercontent.com/122101840/225581983-66f375da-4ad6-4465-9fc5-306ab5ca5067.png)

------------------------
## ResNet50 V2 **WAS A COMPLETE DISASTER**
